lib/queue.js

```diff
@@ -25,276 +48,197 @@ var FetchQueue = function() {
 module.exports = FetchQueue;
 
 FetchQueue.prototype = [];
-FetchQueue.prototype.add = function(protocol, domain, port, path, depth, callback) {
+FetchQueue.prototype.add = function(queueItem, force, callback) {
+    callback = callback instanceof Function ? callback : noop;
+    var queue = this;
 
-    // For legacy reasons
-    if (depth instanceof Function) {
-        callback = depth;
-        depth = 1;
+    if (arguments.length === 2) {
```

Comment:
True, we shouldn't encourage people to call this method directly, so let's always require the same signature